appsec/src/helper/remote_config/listeners/asm_features_listener.cpp

```diff
@@ -52,4 +65,42 @@ void dds::remote_config::asm_features_listener::on_update(const config &config)
         throw error_applying_config(
             "Invalid config json encoded contents: enabled key invalid");
     }
+
+    if (!json_helper::field_exists(ruleset_, "auto_user_instrum")) {
+        service_config_->set_auto_user_instrum(
+            auto_user_instrum_mode::UNDEFINED);
+        return;
```

Comment:
In order to be able to debug if something goes wrong, either exceptions or logging here would be helpful. Here and in the coming returns